File: boat-sample/src/test/java/sample/java/xyz/srclab/core/test/TestSample.java
package sample.java.xyz.srclab.core.test;

import org.testng.annotations.Test;
import xyz.srclab.common.test.TestLogger;
import xyz.srclab.common.test.TestTask;
import xyz.srclab.common.test.Tests;

import java.util.Arrays;

public class TestSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testTests() {
        Tests.testTasks(Arrays.asList(
            TestTask.newTask(() -> {
                logger.log("Run test task!");
            })
        ));
    }
}

File: boat-sample/src/test/kotlin/sample/xyz/srclab/common/test/TestSample.kt
package sample.xyz.srclab.common.test

import org.testng.annotations.Test
import xyz.srclab.common.test.TestLogger
import xyz.srclab.common.test.TestTask
import xyz.srclab.common.test.testTasks

class TestSampleKt {

    @Test
    fun testTests() {
        testTasks(
            listOf(
                TestTask.newTask { logger.log("Run test task!") }
            )
        )
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}